app/metal-controller-manager/controllers/server_controller.go

```diff
@@ -129,6 +145,21 @@ func (r *ServerReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 		}
 	}
 
+	if s.ObjectMeta.DeletionTimestamp.IsZero() {
+		if finalizerIndex(s.ObjectMeta.Finalizers, serverBindingFinalizer) == -1 {
```

Comment:
There are helper methods for that already AddFinalizer, RemoveFinalizer

Reply:
oh, nice, thanks :) This API is a bit in a mess, hard to find things.